Authentication/SignUp/View/SignupConfiguration.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  SignupConfiguration.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 7/15/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+/**
+ Protocol for handling transition events occured during signup.
+ */
+public protocol SignupControllerTransitionDelegate {
+    
+    func onLogin(controller: SignupController)
+    
+}
+
+/**
+ Class for configuring the singup controller.
+ Includes all information required:
+ view factory method,
+ view model,
+ signup controller delegate and
+ signup controller transition delegate.
+ */
+public final class SignupControllerConfiguration {
+    
+    var viewModel: SignupViewModelType
+    var viewFactory: () -> SignupViewType
```

Comment:
why is this var? 
I also think we should declare its accessibility, as as simple sight I don't know if its meant to be public or internal.